src/responder/kcm/secrets/secrets.c

```diff
@@ -233,11 +261,14 @@ static errno_t get_secret_expiration_time(uint8_t *key, size_t key_length,
         goto done;
     }
 
+    /* Look for the (first) expiration time while securely deleting the
+     * unmarshalled credentials. */
     for (cred = cred_list; *cred != NULL; cred++) {
-        if ((*cred)->times.endtime != 0) {
+        if (expiration == 0 && (*cred)->times.endtime != 0) {
             expiration = (time_t) (*cred)->times.endtime;
-            break;
         }
+        sss_erase_talloc_mem_securely(*cred);
+        *cred = NULL;
```

Comment:
Why?

Reply:
This was a left over. I already removed it.

I initially thought it was a good idea to prevent access to zeroed memory, but later dicovered it wasn't such a good idea because the krb5 functions use those pointers to free the memory (they don't use `talloc`). It seems I forgot to remove this last one.